### ISSUE TITLE
Raphnet adapter performance bad

### ISSUE THREAD

**Issue by Errordoc** [m64p/m64p#275]
since v2022.07.6 the performance is worse again in some games, especially Top Gear Rally. In some games, such as F-Zero, you only notice it in conjunction with the Rumble-Pak in the destruction sequence. These problems only occur in conjunction with the Raphnet plugin. With the other plugin everything works as it should but the Rumble-Pak does not work in conjunction with an original N64 controller.

Windows 11
AMD System
Ryzen 5800x
RX 6800

these problems also exist on other systems with other specs

**Comment by loganmc10** [comment]
Yes there was a hack in place to try and make this better, but I removed it since I think it could cause other unintended issues, even without the raphnet adapter

This problem is a problem with the raphnet plugin itself, and also affects PJ64 see:

https://github.com/raphnet/mupen64plus-input-raphnetraw/issues/3
https://github.com/raphnet/mupen64plus-input-raphnetraw/issues/2

The problem that the round trip time to query the controller is very long, and it stalls the CPU while it's happening, so when a game queries the controller at a high frequency, it causes lag. I'll think about possible solutions, I have a raphnet adapter as well.

**Comment by DarthSidious666** [comment]
Has this issue been fixed since?

**Comment by Errordoc** [comment]
Yes the error is still present


________________________________
Von: DarthSidious666 ***@***.***>
Gesendet: Donnerstag, 25. August 2022, 14:26
An: m64p/m64p ***@***.***>
Cc: Errordoc ***@***.***>; Author ***@***.***>
Betreff: Re: [m64p/m64p] Raphnet adapter performance bad (Issue #275)


Has this issue been fixed since?

**Comment by HazNut** [comment]
I'm experiencing lag with the plugin as well, recently got my N64 to USB adapter (v3) and running F-Zero X on simple64, when my car is destroyed the game starts continuously stuttering, but this only happens while the Rumble Pak is connected, if I unplug it, it stops completely. It also seems to happen when I pause the game, as I can hear the audio stuttering when I pause.

#raphnet/mupen64plus-input-raphnetraw/issues/2
